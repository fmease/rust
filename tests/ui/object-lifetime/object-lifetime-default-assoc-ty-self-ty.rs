// FIXME: Explainer.
//@ known-bug: unknown

trait Outer<'a>: 'a { type Ty; }
trait Inner {}

impl<'a> Outer<'a> for dyn Inner + 'a { type Ty = &'a (); }

fn f<'r>(x: <dyn Inner + 'r as Outer<'r>>::Ty) { g(x) }
// FIXME: Deduce `dyn Inner + 'r` from bound `'a` on self ty param of trait `Outer`.
fn g<'r>(x: <dyn Inner as Outer<'r>>::Ty) {}

fn main() {}
