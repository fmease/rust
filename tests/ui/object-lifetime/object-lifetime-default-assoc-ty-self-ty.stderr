error[E0228]: the lifetime bound for this object type cannot be deduced from context; please supply an explicit bound
  --> $DIR/object-lifetime-default-assoc-ty-self-ty.rs:11:14
   |
LL | fn g<'r>(x: <dyn Inner as Outer<'r>>::Ty) {}
   |              ^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0228`.
